doc/Quality_white_wine_predictor.md

```diff
@@ -0,0 +1,245 @@
+using the following algorithms to perform this task:
+
+1.  **DummyRegressor()**: is one that takes only the value 0 or 1 to
+    indicate the absence or presence of some categorical effect that may
+    be expected to shift the outcome.
+2.  **Ridge**: is a model tuning method that is used to analyse any data
+    that suffers from multicollinearity. This method performs L2
+    regularization. When the issue of multicollinearity occurs,
+    least-squares are unbiased, and variances are large, this results in
+    predicted values to be far away from the actual values.
+3.  **Random Forest**: is an ensemble learning method for
+    classification, regression and other tasks that operates by
+    constructing a multitude of decision trees at training time. For
+    classification tasks, the output of the random forest is the class
+    selected by most trees. For regression tasks, the mean or average
+    prediction of the individual trees is returned
+4.  **KNN**: is an approach to data classification that estimates how
+    likely a data point is to be a member of one group or the other
+    depending on what group the data points nearest to it are in.The
+    k-nearest-neighbor is an example of a “lazy learner” algorithm,
+    meaning that it does not build a model using the training set until
+    a query of the data set is performed.
+5.  **Bayes**: is an algorithm that uses Bayes’ theorem to classify
+    objects. Naive Bayes classifiers assume strong, or naive,
+    independence between attributes of data points. Popular uses of
+    naive Bayes classifiers include spam filters, text analysis and
+    medical diagnosis.
+6.  **SVM**: is machine learning algorithm that analyzes data for
+    classification and regression analysis. SVM is a supervised learning
+    method that looks at data and sorts it into one of two categories.
+    An SVM outputs a map of the sorted data with the margins between the
+    two as far apart as possible.
+
+# Analysis
+
+## **Data Cleaning and Preprocessing**
+
+The first step clean and prepare the data for analysis. First, It is
+necessary to checked the data types focusing on numerical and
+categorical to simplify the correlation’s computation and visualization.
+Second, it’s necessary to identify any missing values existing in our
+data set. Last, it’s relevant to research each column/feature’s
+statistical summary to detect any problem like outliers and abnormal
+distributions.
+
+After, data preprocessing is crucial in any data mining process as they
+directly impact success rate of the project. This reduces complexity of
+the data under analysis as data in real world is unclean. we split the
+dataset in two parts, one for training and one for testing. The model
+building and tuning is done in the training set, and then we use the
+test set to predict new values and evaluate the results.
+
+## **EDA first conclusions**
+
+According to our first EDA, we do not have a balanced database, our
```

Comment:
The points made in this paragraph are repeated in the following ones. Maybe it's better to keep this introduction shorter?